### DISCUSSION TITLE
Where does brew set $PATH on OSX (mac)?

### DISCUSSION THREAD

**Discussion by mecoookie** [Homebrew/discussions#4452] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 763c41f006ab9b44261e313b0aaa24a047431b06
Last commit: 15 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7ee397e828fc2b17eff92e614a8a0d2e5417419c
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 24 Apr 22:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: N/A
Xcode: 14.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Where does brew modify the $PATH on OSX (Mac)? 

This is the output of my path directory:

 % echo $PATH
/Users/josh/opt/anaconda3/bin:/Users/josh/opt/anaconda3/condabin:/usr/local/opt/python/libexec/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/Little Snitch.app/Contents/Components:/Library/Apple/usr/bin

I have brew installed with conda, and I'm finding conda puts it's paths in the front head of homebrew, but  I can't find where brew actually sets it's path variables! 

I checked my .zshrc, .bash_profile, /etc/paths, and /etc/paths.d.
I can't find anywhere that the Brew directories are adde to $PATH. Is it a new setting in the system somewhere? or a file I don't know?

I want to make sure brew commands are taking preference, at least during testing, so I'd like to set my path so the brew directories come first.

**Comment by carlocab** [comment]
You probably have an
```
eval "$(/opt/homebrew/bin/brew shellenv)"
```
somewhere in your config files that does this.

**Comment by cho-m** [comment]
If you copy-pasted output of Homebrew installer, then it will be in:
- `.zprofile` for ZSH (default on macOS)
- `.bash_profile` or `.profile` if you changed shell to Bash
- `.profile` everywhere else (again if you changed shell to something like csh/tsch/ksh/...)

See: https://github.com/Homebrew/install/blob/master/install.sh#L992-L1005

---

For interactive+login ZSH terminal, your shell will load `.zprofile` before `.zshrc`, so that may impact `PATH` order if prepended in both files.

Other config files also exist that you may want to check.

See: https://zsh.sourceforge.io/Doc/Release/Files.html

**Comment by JosephTLyons** [comment]
@mecoookie, did you ever find out where brew sets the $PATH? I'm looking everywhere on my end and can't find a single thing setting it - no idea how it is ending up in the $PATH.

**Reply by gromgit** [reply]
I've answered this question enough times that I decided to write a script for folks to use. `brew tap gromgit/brewtils`, then `brew trace-login` to run a complete traced login process that logs to `/tmp/trace.log` by default.

`brew trace-login --help` to see the various options available. For instance, `brew trace-login -- -w PATH=` will print just the lines that set `PATH`.
